compiler/damlc/tests/daml-test-files/ConstrainedClassMethod.daml

```diff
@@ -9,5 +7,8 @@ class A t where
     foo : t -> t
     bar : Eq t => t -> t
 
-class B t where
-    baz : B b => b -> t
+-- This would create a recursive type synonym.
```

Comment:
We decided to just disable this for now. We don’t use this on our large internal DAML codebase. There is a reasonable workaround with manual dictionary passing and if it does become a record we could either handle this in the LF conversion or support infinite types (something like OCaml’s `-rectypes`).

Reply:
Seems reasonable to me. I don't know of any natural examples of a class that requires itself as a constraint.